controller/platform_clients.go

```diff
@@ -0,0 +1,174 @@
+package controller
+
+import (
+	"context"
+	"github.com/fabric8-services/fabric8-wit/configuration"
+	"github.com/fabric8-services/fabric8-wit/errors"
+	"github.com/fabric8-services/fabric8-wit/kubernetes"
+	"github.com/fabric8-services/fabric8-wit/log"
+	"github.com/goadesign/goa"
+	errs "github.com/pkg/errors"
+	"net/url"
+	"os"
+)
+
+// ClientGetter creates an instances of clients used by this controller
+type ClientGetter interface {
+	GetKubeClient(ctx context.Context) (kubernetes.KubeClientInterface, error)
+	GetAndCheckOSIOClient(ctx context.Context) (OpenshiftIOClient, error)
+	GetOSClient(ctx context.Context) (kubernetes.OpenShiftRESTAPI, error)
+}
+
+// Default implementation of OSClientGetter and OSIOClientGetter used by NewPipelineController
+type defaultClientGetter struct {
+	config *configuration.Registry
+}
+
+func (g *defaultClientGetter) GetAndCheckOSIOClient(ctx context.Context) (OpenshiftIOClient, error) {
+
+	// defaults
+	host := "localhost"
+	scheme := "https"
+
+	req := goa.ContextRequest(ctx)
+	if req != nil {
+		// Note - it's probably more efficient to force a loopback host, and only use the port number here
+		// (on some systems using a non-loopback interface forces a network stack traverse)
+		host = req.Host
+		scheme = req.URL.Scheme
+	}
+
+	// The deployments API communicates with the rest of WIT via the stnadard WIT API.
+	// This environment variable is used for local development of the deployments API, to point ot a remote WIT.
+	witURLStr := os.Getenv("FABRIC8_WIT_API_URL")
+	if witURLStr != "" {
+		witurl, err := url.Parse(witURLStr)
+		if err != nil {
+			log.Error(ctx, map[string]interface{}{
+				"FABRIC8_WIT_API_URL": witURLStr,
+				"err": err,
+			}, "cannot parse FABRIC8_WIT_API_URL: %s", witURLStr)
+			return nil, errs.Wrapf(err, "cannot parse FABRIC8_WIT_API_URL: %s", witURLStr)
```

Comment:
Could you please use a more specific error from the `errors` package. That way, `jsonapi.JSONErrorResponse(ctx, err)` will convert it into the right status code.